app/jobs/process_delius_data_job.rb

```diff
@@ -62,6 +62,10 @@ def process_record(delius_record)
     end
 
     update_com_name(delius_record)
+
+    if case_information.saved_changes?
```

Comment:
could we not just invert the 'unless' into an if - so it becomes if case.information.save <nDelius> else <create error>? Also inlining this process_record method might lead down the path of just calling nDelius inline as we have the offender object in our hands?